src/igor/extend.go

```diff
@@ -0,0 +1,143 @@
+// Copyright (2013) Sandia Corporation.
+// Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
+// the U.S. Government retains certain rights in this software.
+
+package main
+
+import (
+	"fmt"
+	log "minilog"
+	"ranges"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var cmdExtend = &Command{
+	UsageLine: "extend -r <reservation name> -t <time>",
+	Short:     "extend a reservation",
+	Long: `
+Extend an existing reservation.
+
+REQUIRED FLAGS:
+
+The -r flag specifies the name of the existing reservation.
+
+The -t flag is used to specify the reservation extension time. Time denominations should
+be specified in days(d), hours(h), and minutes(m), in that order. Unitless
+numbers are treated as minutes. Days are defined as 24*60 minutes. Example: To
+make a reservation for 7 days: -t 7d. To make a reservation for 4 days, 6
+hours, 30 minutes: -t 4d6h30m (default = 60m).`,
+}
+
+func init() {
+	// break init cycle
+	cmdExtend.Run = runExtend
+
+	cmdExtend.Flag.StringVar(&subR, "r", "", "")
+	cmdExtend.Flag.StringVar(&subT, "t", "60m", "")
+}
+
+func runExtend(cmd *Command, args []string) {
+	// duration is in minutes
+	duration := 0
+
+	v, err := strconv.Atoi(subT)
```

Comment:
Pull this out into a function and call it from `runSub` and `runExtend` rather than copy it.

```
// parseDuration parses a duration, supporting a 'd' suffix in addition to those supported by time.ParseDuration.
// Returns the duration in minutes.
func parseDuration(s string) (int, error) {
...
}
```

Reply:
Where should the resultant function go? In main.go?

Reply:
Either `time.go` or `misc.go`.